CMakeLists.txt

```diff
@@ -34,18 +34,17 @@ install(EXPORT ArborXTargets
 )
 
 SET(ARBORX_VERSION_STRING "0.9 (dev)")
-SET(ARBORX_GIT_COMMIT_HASH "No hash available")
-IF(EXISTS ${PROJECT_SOURCE_DIR}/.git)
-    FIND_PACKAGE(Git)
-    IF(GIT_FOUND)
-        EXECUTE_PROCESS(
-            COMMAND           ${GIT_EXECUTABLE} log --pretty=format:%h -n 1
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-            OUTPUT_VARIABLE   ARBORX_GIT_COMMIT_HASH
-            )
-    ENDIF()
-ENDIF()
-MESSAGE("ArborX hash = '${ARBORX_GIT_COMMIT_HASH}'")
+
+# Makes sure to store the current git hash ArborX_Version.hpp each time we 
+# recompile. 
+ADD_CUSTOM_TARGET(
+  record_hash ALL VERBATIM
+  COMMAND ${CMAKE_COMMAND}
+    -DSOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
+    -DBINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}
+    -DARBORX_VERSION_STRING=${ARBORX_VERSION_STRING}
+    -P cmake/SetupVersion.cmake
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
I don't think that this does what we want. Because now if I install ArborX using simply `make install`, I don't think that `ArborX_Version.hpp` will exist and so there is no way to get the version using a runtime command which is against xSDK guidelines.

Reply:
`ArborX_Version.hpp` is installed.

Reply:
But is it going to be installed even if I never call `make record_hash`? Because in that case I never call `configure_file`.

Reply:
From the `CMake` documentation for `add_custom_target`:
```
ALL
    Indicate that this target should be added to the default build target so that it will be run every time (the command cannot be called ALL).
```
The target will run by default when you just do `make`.